keras_nlp/models/generative_task.py

```diff
@@ -234,23 +236,39 @@ def generate(
                 `preprocessor`. If `preprocessor` is `None`, `inputs` should be
                 should be padded to the desired maximum length and this argument
                 will be ignored.
+            stop_token_ids: Optional. None, "auto", or tuple of token ids. Defaults
+                to "auto" which uses the `self.preprocessor.tokenizer.end_token_id`.
```

Comment:
Let's not refer to self in a docstring. Maybe just say `preprocessor.tokenizer.end_token_id`.